quota-service/src/modules/user/userService.js

```diff
@@ -43,9 +44,10 @@ class UserService {
   }
 
   async consumeQuota(key, consumed) {
+    const { consumedQuota = 0 } = await this.getCollection().findOne({ key });
```

Comment:
Mongo already has a function to add without retrieving first, check:
https://docs.mongodb.com/manual/reference/operator/update/inc/